src/main/java/az/gdg/msauth/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,187 @@
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.UUID;
+
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    private final UserRepository userRepository;
+    private final TokenUtil tokenUtil;
+    private final AuthenticationService authenticationService;
+    private final EmailService emailService;
+    private static final Logger logger = LoggerFactory.getLogger(UserServiceImpl.class);
+
+    public UserServiceImpl(UserRepository userRepository, AuthenticationService authenticationService,
+                           EmailService emailService, TokenUtil tokenUtil) {
+        this.userRepository = userRepository;
+        this.authenticationService = authenticationService;
+        this.emailService = emailService;
+        this.tokenUtil = tokenUtil;
+    }
+
+    public void signUp(UserDTO userDTO) {
+        logger.info("ActionLog.Sign up user.Start");
+
+        UserEntity checkedEmail = userRepository.findByEmail(userDTO.getEmail());
+        if (checkedEmail != null) {
+            logger.error("ActionLog.WrongDataException.Thrown");
+            throw new WrongDataException("This email already exists");
+        }
+
+        String password = new BCryptPasswordEncoder().encode(userDTO.getPassword());
+        String code = UUID.randomUUID().toString();
+        UserEntity userEntity = UserEntity
+                .builder()
+                .name(userDTO.getName())
+                .surname(userDTO.getSurname())
+                .username(userDTO.getEmail())
+                .email(userDTO.getEmail())
+                .password(password)
+                .accountVerificationCode(code)
+                .role(Role.ROLE_USER)
+                .status(Status.REGISTERED)
+                .build();
+
+        userRepository.save(userEntity);
+
+        MailDTO mail = new MailDTO().builder()
+                .mailTo(Collections.singletonList(userDTO.getEmail()))
+                .mailSubject("Your registration letter")
+                .mailBody("<h2>" + "Verify Account" + "</h2>" + "</br>" +
+                        "<a href=" +
+                        "https://gdg-ms-auth.herokuapp.com/user/verify-account?email=" + userDTO.getEmail() +
+                        "&code=" + code + ">" +
+                        "https://gdg-ms-auth.herokuapp.com/user/verify-account?email=" + userDTO.getEmail() +
+                        "&code=" + code + "</a>")
+                .build();
+
+        emailService.sendToQueue(mail);
+        logger.info("ActionLog.Sign up user.Stop.Success");
+
+    }
+
+    public String getCustomerIdByEmail(String token, String email) {
+        logger.info("ActionLog.GetCustomerIdByEmail.Start");
+        UserInfo userInfo = authenticationService.validateToken(token);
+        String userRole = userInfo.getRole();
+        if (!userRole.equals("ROLE_ADMIN")) {
+            logger.error("ActionLog.AuthenticationException.Thrown");
```

Comment:
You are using userRole only in this if statement, why do you need to create a variable for it ?